tests/puppeteer/campaigns/functional/BO/11_international/01_localization/04_defaultCurrency.js

```diff
@@ -0,0 +1,169 @@
+    });
+
+    it('should import localization pack', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'importLocalizationPack', baseContext);
+      const textResult = await this.pageObjects.localizationPage.importLocalizationPack('Chile', contentToImport);
+      await expect(textResult).to.equal(this.pageObjects.localizationPage.importLocalizationPackSuccessfulMessage);
+    });
+  });
+
+  const tests = [
+    {
+      args: {
+        defaultCurrency: `${Currencies.chileanPeso.name} (${Currencies.chileanPeso.isoCode})`,
+        currency: `${Currencies.chileanPeso.isoCode} ${Currencies.chileanPeso.symbol}`,
+      },
+    },
+    {
+      args: {
+        defaultCurrency: `${Currencies.euro.name} (${Currencies.euro.isoCode})`,
+        currency: `${Currencies.euro.isoCode} ${Currencies.euro.symbol}`,
+      },
+    },
+  ];
+  tests.forEach((test, index) => {
+    describe(`Choose default currency '${test.args.defaultCurrency}' and check it in FO`, async () => {
+      before(async function () {
+        browser = await helper.createBrowser();
+        page = await helper.newTab(browser);
+        this.pageObjects = await init();
+      });
+      after(async () => {
+        await helper.closeBrowser(browser);
+      });
+
+      loginCommon.loginBO();
+
+      it('should go to localization page', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          'goToLocalizationPageToChooseDefaultCurrency',
+          baseContext,
+        );
+        await this.pageObjects.boBasePage.goToSubMenu(
+          this.pageObjects.boBasePage.internationalParentLink,
+          this.pageObjects.boBasePage.localizationLink,
+        );
+        await this.pageObjects.boBasePage.closeSfToolBar();
+        const pageTitle = await this.pageObjects.localizationPage.getPageTitle();
+        await expect(pageTitle).to.contains(this.pageObjects.localizationPage.pageTitle);
+      });
+
+      it('should choose default currency', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `setDefaultCurrency${index}`, baseContext);
+        const textResult = await this.pageObjects.localizationPage.setDefaultCurrency(test.args.defaultCurrency);
+        await expect(textResult).to.equal('Update successful');
+      });
+
+      it('should go to FO and check the existence of currency', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `checkCurrencyInFO${index}`, baseContext);
+        page = await this.pageObjects.boBasePage.viewMyShop();
+        this.pageObjects = await init();
+        const defaultCurrency = await this.pageObjects.foBasePage.getDefaultCurrency();
+        await expect(defaultCurrency).to.equal(test.args.currency);
+        page = await this.pageObjects.foBasePage.closePage(browser, 1);
+        this.pageObjects = await init();
+      });
+
+      if (index === 1) {
```

Comment:
What were you trying to do here ? Is there a better way to handle this ?

Reply:
The condition is used to do the deletion of the currency during the last iteration, I have no solution because the browser will be closed, so the describe should be inside the forEach method. 
What do you think ?

Reply:
This section must be run only for the *last* currency, or for *every currency but the first* ?

Reply:
The section must be run at the end of the test, it's not related to currency.

Reply:
```suggestion
      if (index === (tests.length - 1)) {
```
Maybe you should use this, to make sure it doesn't break if you add a currency in the `tests` objects.
Please test because I don't remember if `index` starts at 0 or 1 (Javascript....).

Reply:
I will fix it :) thank you